src/test/suite/didactUriCompletionItemProvider.test.ts

```diff
@@ -42,13 +49,56 @@ suite("Didact URI completion provider tests", function () {
 		});
 	});
 
-	test("that the didact protocol completion returns with didact://?commandId=", () => {
-		const completionItem = provider.didactProtocolCompletion();
+	test("that the didact protocol completion returns with didact://?commandId=", async () => {
+		const textDocument: vscode.TextDocument | undefined = await vscode.workspace.openTextDocument();
+		const position = new vscode.Position(0,0);
+		const completionItem = provider.didactProtocolCompletion(textDocument, position);
 		expect(completionItem).to.not.be.null;
 		expect(completionItem).to.not.be.undefined;
 		expect(completionItem.insertText).to.be.equal(DIDACT_COMMAND_PREFIX);
 	});
 
+	test("that completions work the way they should", async () => {
+		const listOfCompletions : string[] = [
+			'[My Link](didact:',
+			'[My Link](didact:/',
+			'[My Link](didact://',
+			'[My Link](didact://?',
+			'[My Link](didact://?c',
+			'[My Link](didact://?co',
+			'[My Link](didact://?com'
+		];
+		const expected = "[My Link](didact://?commandId=";
+
+		suite('walk through each provided completion', () => {
+			listOfCompletions.forEach(function(stringToTest: string){
+				before( async () => {
+					await delay(2000);
```

Comment:
delay to replace with conditional wait

Reply:
can you clarify what you mean here as far as a conditional wait?

Reply:
using a delay in test is unreliable and slow. It is better to use an active check

for instance, we are using https://www.npmjs.com/package/async-wait-until in several of our tests

Reply:
though I appreciate what you're after here, I am going to leave the delays for now and we can take another look in a separate pass. if it was showing up as flaky, I would go further. I did take a look to see if I could quickly make it work and due to the chained series of command executions it was not an easy transition to using waituntil.

if down the line you want to explore using waituntil, I'm all for it.

Reply:
Will investigate the conditional wait change separately -- documented in the JIRA - https://issues.redhat.com/browse/FUSETOOLS2-869?focusedCommentId=15505352&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15505352